sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Services/WebPubSubRequestExtensions.cs

```diff
@@ -152,11 +152,10 @@ internal static bool IsValidSignature(this WebPubSubConnectionContext connection
             {
                 var states = new Dictionary<string, object>();
                 var parsedStates = Encoding.UTF8.GetString(Convert.FromBase64String(connectionStates));
-                var statesObj = JsonDocument.Parse(parsedStates);
+                using var statesObj = JsonDocument.Parse(parsedStates);
                 foreach (var item in statesObj.RootElement.EnumerateObject())
                 {
-                    // Use ToString() to set pure value without ValueKind.
-                    states.Add(item.Name, item.Value.ToString());
```

Comment:
@Mohit-Chakraborty The change here is probably a break change require customer to process the value differently.